src/ngraph/pass/constant_folding_one_hot.cpp

```diff
@@ -0,0 +1,226 @@
+        {
+        case element::Type_t::undefined:
+            NGRAPH_CHECK(false, "Encountered 'undefined' element type in one_hot_callback");
+            break;
+        case element::Type_t::dynamic:
+            NGRAPH_CHECK(false, "Encountered 'dynamic' element type in one_hot_callback");
+            break;
+        case element::Type_t::u1:
+            NGRAPH_CHECK(false, "Encountered 'u1' element type in one_hot_callback");
+            break;
+        case element::Type_t::boolean:
+            replacement =
+                fold_constant_one_hot<char>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::bf16:
+            replacement = fold_constant_one_hot<bfloat16>(
+                indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::f16:
+            replacement =
+                fold_constant_one_hot<float16>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::f32:
+            replacement =
+                fold_constant_one_hot<float>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::f64:
+            replacement =
+                fold_constant_one_hot<double>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::i8:
+            replacement =
+                fold_constant_one_hot<int8_t>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::i16:
+            replacement =
+                fold_constant_one_hot<int16_t>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::i32:
+            replacement =
+                fold_constant_one_hot<int32_t>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::i64:
+            replacement =
+                fold_constant_one_hot<int64_t>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::u8:
+            replacement =
+                fold_constant_one_hot<uint8_t>(indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::u16:
+            replacement = fold_constant_one_hot<uint16_t>(
+                indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::u32:
+            replacement = fold_constant_one_hot<uint32_t>(
+                indices_node, on_node, off_node, output_shape, axis);
+            break;
+        case element::Type_t::u64:
+            replacement = fold_constant_one_hot<uint64_t>(
+                indices_node, on_node, off_node, output_shape, axis);
+            break;
+        }
+
+        replace_node(m.get_match_root(), replacement);
+        return true;
+    };
+    auto split_matcher =
```

Comment:
minor nit. Change name to onehot_matcher.

Reply:
Done